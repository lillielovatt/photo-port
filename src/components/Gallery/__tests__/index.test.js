import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Gallery from "..";

// will require a prop of currentCategory
// mock it here, instead of taking it from App component

const portrait = {
    name: "portraits",
    description: "Portraits of people in my life",
};

afterEach(cleanup);

describe("Gallery is rendering", () => {
    it("renders", () => {
        render(<Gallery currentCategory={portrait} />);
    });
    it("renders", () => {
        const { getByTestId } = render(<Gallery currentCategory={portrait} />);
        expect(getByTestId("h1tag")).toHaveTextContent("Portraits");
    });
});

describe("Gallery snapshot", () => {
    it("matches snapshot", () => {
        const { asFragment } = render();
        expect(asFragment()).toMatchSnapshot();
    });
});
