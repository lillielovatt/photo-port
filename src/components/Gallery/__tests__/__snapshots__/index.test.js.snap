// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery snapshot matches snapshot 1`] = `<DocumentFragment />`;
